### PR TITLE
Query builder

### PR DESCRIPTION
**Pull request by msbit**
Translate the built-up SQL strings to using query builder.

The relevant SQL executed before:

```
select * from craft_commerceregisteroncheckout where orderNumber='366dda7bb6b6ffb53116bcaadd4195d8' ORDER BY dateUpdated DESC

delete from craft_commerceregisteroncheckout where orderNumber='366dda7bb6b6ffb53116bcaadd4195d8'

delete from craft_commerceregisteroncheckout where dateUpdated<='2016-12-16 18:33:52'

select * from craft_commerce_customers where email='user2@email.com' ORDER BY dateUpdated DESC
```

and after:

```
SELECT *
FROM `craft_commerceregisteroncheckout`
WHERE `orderNumber`='dc49f09f6d030494023718b3f5c6311d'
ORDER BY `dateUpdated` DESC

DELETE FROM `craft_commerceregisteroncheckout` WHERE `orderNumber`='dc49f09f6d030494023718b3f5c6311d'

DELETE FROM `craft_commerceregisteroncheckout` WHERE dateUpdated <= '2016-12-16 18:30:38'

SELECT *
FROM `craft_commerce_customers`
WHERE `email`='user1@email.com'
ORDER BY `dateUpdated` DESC
```

As part of this, `$tableName` was no longer referred to, so removed this also.

### CODE REVIEW

**Comment by bossanova808** [comment]
Thanks Tom

For simple stuff I almost prefer the straight SQL approach TBH, I find the query builder a bit wordy.

Still, I suppose if it is equivalent it's much of a muchness.  And I presume you've got some plans to add stuff, e.g. support for those extra fields or whatever?

**Comment by msbit** [comment]
@bossanova808 ah no, mainly just a quick update to match with uses of `craft()->db->createCommand()->update(...)` throughout, and to remove manually generated SQL strings.
I can live with it not being merged, if you prefer to keep the straightforward SQL requests :)

**Comment by bossanova808** [comment]
I'm just being old fashioned.  I'll merge it but might wait for something else to come along before doing a new version as such

**Comment by bossanova808** [comment]
(You've tested it right?)

**Comment by msbit** [comment]
@bossanova808 tested it explicitly while logging the SQL queries as listed above, and I've had the branch checked out in place on my development machine while running through the cart->registration flow without issue.

Having said that, I definitely feel your concern as maintainer about the coverage of the tests, and don't want to push you into an uncomfortable merge. Given that this amounts to no more than syntactic sugar, it can sit around for a bit, maybe I could look at some unit testing if I get sufficiently motivated.

**Comment by bossanova808** [comment]
This really looks fine, and it is neater and more Crafty, so going to whack it in
